src/libopensc/card-myeid.c

```diff
@@ -918,25 +922,46 @@ int myeid_ecdh_derive(struct sc_card *card, const u8* pubkey, size_t pubkey_len,
 
 	struct sc_apdu apdu;
 	u8 sbuf[SC_MAX_APDU_BUFFER_SIZE];
-	u8 rbuf[SC_MAX_APDU_BUFFER_SIZE];
+	u8 rbuf[SC_MAX_APDU_BUFFER_SIZE];	
 
 	int r;
+	size_t ext_len_bytes;
 
 	sc_format_apdu(card, &apdu,
 	        SC_APDU_CASE_4_SHORT,
 	        0x86, 0x00, 0x00);
 
 	apdu.resp = rbuf;
 	apdu.resplen = sizeof(rbuf);
-
-	/* Fill in "Data objects in dynamic authentication template (tag 0x7C) structure */
+	
+	/* Fill in "Data objects in dynamic authentication template" (tag 0x7C) structure 
+	*
+	*  Size of the structure depends on key length. With 521 bit keys two bytes are needed for defining length of a point.
+	*/		
+	
 	sbuf[0] = 0x7C;
-	sbuf[1] = pubkey_len + 4;
-	sbuf[2] = 0x85;
-	sbuf[3] = pubkey_len;
-	memcpy(&sbuf[4], pubkey, pubkey_len);
+	ext_len_bytes = 0;
 
-	apdu.lc = pubkey_len + 4;
+	if (pubkey_len > 127)
```

Comment:
This would be better, but it might take a while before I will have the next possibility to get familiar with the asn1 functions and refactor this. Would it be possible to make this a TODO and merge this implementation to fix non working code?